pytorch_lightning/callbacks/base.py

```diff
@@ -0,0 +1,90 @@
+"""
+Callbacks
+=========
+
+Callbacks supported by Lightning
+"""
+
+import abc
+
+
+_NO_TRAINER_ERROR_MSG = ".set_trainer() should be called after the callback initialization"
+
+
+class Callback(abc.ABC):
+    """Abstract base class used to build new callbacks."""
+
+    def __init__(self):
+        self._trainer = None
+
+    @property
+    def trainer(self):
+        assert self._trainer is not None, _NO_TRAINER_ERROR_MSG
+        return self._trainer
+
+    @property
+    def default_save_path(self):
```

Comment:
do we need this method which just wraps one more call?

Reply:
No, we don't.

My intention here was to encourage callback's creators to use this instead of using a `dir_path` argument in the init of the callback so all the files stay in the trainer's `default_save_path`. 

I can remove it if you want.